docs/OVERVIEW.md

```diff
@@ -0,0 +1,155 @@
+
+* Alice extracts the confidential identites from the list of participants by filtering those unknown to the `NetworkMapCache`
+* Alice sends the list of confidential identities to Charlie 
+* Charlie attempts to lookup the `Party` using the owning `PublicKey`
+
+`serviceHub.identityService.wellKnownPartyFromAnonymous(newKeyForBob)`
+
+* Any parties that Charlie cannot resolve are sent back to Alice 
+* Alice looks up the parties on her `IdentityService` and sends the owning `PublicKey` and `Party` object back to charlie 
+* Charlie then initiates a `VerifyAndAddKey` flow with each of those parties
+* The security protocol is exactly the same as in `RequestKeyFlow` just without the generation of a new key
+
+
+Once the flow has completed, Charlie will then have a mapping between the owning `PublicKey` of the confidential identity and the `Party` associated with it in his `IdentityService`. The other constructor that takes a `List` of `AnonymousParty`'s can be used when there is a state on the ledger containing a confidential identity but no transaction present. 
+
+ ### Using Confidential Identities with Accounts
+ 
+This new version of confidential identities was written with integration with accounts in mind. The new API on `KeyManagementService` allows us to supply an account identifier, classified by a `UUID`, as a parameter. 
+
+`keyManagementService.freshKey(uuid)`
+
+This API persists a mapping between the new `PublicKey` and the account identitifier in the `PublicKeyHashToExternalId`. There's are two new API's on `IdentityService` for use with accounts:
+1. `externalIdForPublicKey(publicKey: PublicKey) : UUID?`
+2. `publicKeysForExternalId(externalId: UUID): Iterable<PublicKey>` 
+
+These methods provide a mechanism to lookup an account identifier associated with a confidential identity or lookup the confidential `PublicKey`'s associated with a given account identifier.
+
+
+### Example Usage 
+
+Consider a simple `IOUFlow` between two parties in which we would want to utilise confidentital identities. It has one output state:
+`IOUState(value: Int, lender: AbstractParty, borrower: AbstractParty)`
+The flow and counter-flow would be as follows:
+
+```
+@InitiatingFlow
+@StartableByRPC
+class IOUFlow(val iouValue: Int, val otherParty: Party) : FlowLogic<Unit>() {
+    /** The progress tracker provides checkpoints indicating the progress of the flow to observers. */
+    override val progressTracker = ProgressTracker()
+    
+     @Suspendable
+    override fun call() {
+        // We retrieve the notary identity from the network map.
+        val notary = serviceHub.networkMapCache.notaryIdentities[0]
+
+        // We create a confidential identity for ourself acting as the lender
+        val anonymousLender = AnonymousParty(serviceHub.keyManagementService.freshKey())
+        
+        // Register the key in our identity service
+        serviceHub.identityService.registerKey(anonymousMe.owningKey, ourIdentity)
```

Comment:
here, the counter-party won't have stored this key, so maybe we need to do request key on both sides! Originally there was a swap identities flow for this but we can just do it with two calls to request new key from each side of the flow.